File: packages/android/app/src/main/java/com/billyrinks/antimasking/domain/repository/AntiMaskingRepository.kt
package com.billyrinks.antimasking.domain.repository

import com.billyrinks.antimasking.domain.model.CallVerification
import com.billyrinks.antimasking.domain.model.FraudAlert
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for anti-masking operations
 */
interface AntiMaskingRepository {

    /**
     * Verify a call between caller and callee numbers
     */
    suspend fun verifyCall(
        callerNumber: String,
        calleeNumber: String
    ): Result<CallVerification>

    /**
     * Get verification history with optional filtering
     */
    suspend fun getVerifications(
        limit: Int = 20,
        offset: Int = 0,
        maskingDetected: Boolean? = null
    ): Result<List<CallVerification>>

    /**
     * Get a specific verification by ID
     */
    suspend fun getVerificationById(id: String): Result<CallVerification>

    /**
     * Subscribe to real-time verifications
     */
    fun observeVerifications(): Flow<CallVerification>

    /**
     * Report masking to NCC
     */
    suspend fun reportMasking(
        verificationId: String,
        additionalInfo: String? = null
    ): Result<Unit>

    /**
     * Get fraud alerts
     */
    suspend fun getFraudAlerts(
        limit: Int = 20,
        isAcknowledged: Boolean? = null
    ): Result<List<FraudAlert>>

    /**
     * Subscribe to real-time fraud alerts
     */
    fun observeFraudAlerts(): Flow<FraudAlert>

    /**
     * Acknowledge a fraud alert
     */
    suspend fun acknowledgeFraudAlert(alertId: String): Result<Unit>
}

File: packages/flutter/lib/features/anti_masking/domain/repositories/anti_masking_repository.dart
import 'package:dartz/dartz.dart';
import '../../../../core/errors/failures.dart';
import '../entities/call_verification.dart';

/// Anti-masking repository interface
abstract class AntiMaskingRepository {
  /// Verify a call for masking
  Future<Either<Failure, CallVerification>> verifyCall({
    required String callerNumber,
    required String calleeNumber,
  });

  /// Get verification history
  Future<Either<Failure, List<CallVerification>>> getVerificationHistory({
    int page = 1,
    int pageSize = 20,
    bool? maskingOnly,
  });

  /// Get a specific verification by ID
  Future<Either<Failure, CallVerification>> getVerification(String id);

  /// Report masking to NCC
  Future<Either<Failure, bool>> reportMasking({
    required String verificationId,
    required String description,
    String? additionalNotes,
  });

  /// Get fraud alerts
  Future<Either<Failure, List<FraudAlert>>> getFraudAlerts({
    int page = 1,
    int pageSize = 20,
    AlertStatus? status,
    AlertSeverity? minSeverity,
  });

  /// Acknowledge a fraud alert
  Future<Either<Failure, FraudAlert>> acknowledgeAlert(String alertId);

  /// Resolve a fraud alert
  Future<Either<Failure, FraudAlert>> resolveAlert({
    required String alertId,
    required String resolution,
  });

  /// Stream of new fraud alerts (real-time)
  Stream<FraudAlert> watchFraudAlerts();

  /// Stream of new verifications (real-time)
  Stream<CallVerification> watchVerifications();
}
